lizrd/core/llm.py

```diff
@@ -433,7 +425,6 @@ def __init__(self, embedding_layer, encoder_tower, head):
                 ]
             )
         )
-        self.head = head
```

Comment:
Maybe we can just have `self.embedding_layer`, `self.encoder_tower`, `self.head`  instead of `self.full_model`?

Reply:
Then we also need to change forward

Reply:
But actually that actually that should look better than this

Reply:
It's unusual to create an `nn.Module` on each forward. I don't know if something might break because of that, but kinda looks like that. Let's resolve this conversation if there is we have no other idea how to do what we do here.